### PR TITLE
Validate the list of groups before updating the user claims

### PR DESCRIPTION
**Pull request by somindatommy**
### Purpose
Break the user claim update flow if the updated groups do not exist in the system.

### CODE REVIEW

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java [hunk 1]
```diff
@@ -67,6 +69,7 @@
 public class SCIMUserOperationListener extends AbstractIdentityUserOperationEventListener {
 
     private static final Log log = LogFactory.getLog(SCIMUserOperationListener.class);
+    private static final String VALUE_SEPARATOR = ",";
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java:72]: shall we use the MultiAttributeSeperator constant in kernel?

Comment (somindatommy) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java]: Fixed with 06768cde82fd685e208bdf2a3fdbbaaa5649ec71

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java [hunk 2]
```diff
@@ -313,6 +325,70 @@ private void validateClaimUpdate(String username) throws UserStoreException {
...
+        if (StringUtils.isEmpty(attributeSeparator)) {
+            attributeSeparator = DEFAULT_VALUE_SEPARATOR;
+        }
+        // We need to split if the user has provided a list of groups.
+        String[] groups = value.split(attributeSeparator);
+        boolean hasInvalidGroups = false;
+        for (String groupName : groups) {
+            // We need to identify the groups that does not exist in the system.
+            if (!userStoreManager.isExistingRole(groupName)) {
+                hasInvalidGroups = true;
```
Comment (emswbandara) [inline components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java:383]: shall we add a debug log here logging the invalid group names.

**Review by emswbandara — APPROVED** [review]

File: components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java (hunk 2 above)
Comment (somindatommy) [reply components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java:392]: Fixed with a34f6b065f6b4f2316f6dce8ff05bb6262eae82f

**Pull request merged**